rust/gui-client/src-common/src/ipc.rs

```diff
@@ -1,54 +1,31 @@
 use anyhow::{Context as _, Result};
 use firezone_headless_client::{
     ipc::{self, Error},
-    IpcClientMsg, IpcServerMsg,
+    IpcClientMsg,
 };
-use futures::{SinkExt, StreamExt};
+use futures::SinkExt;
 use secrecy::{ExposeSecret, SecretString};
 use std::net::IpAddr;
 
-pub enum Event {
-    Closed,
-    Message(IpcServerMsg),
-    ReadFailed(anyhow::Error),
-}
+pub use firezone_headless_client::ipc::ClientRead;
 
 pub struct Client {
-    task: tokio::task::JoinHandle<Result<()>>,
     // Needed temporarily to avoid a big refactor. We can remove this in the future.
     tx: ipc::ClientWrite,
 }
 
-impl Drop for Client {
-    // Might drop in-flight IPC messages
-    fn drop(&mut self) {
-        self.task.abort();
-    }
-}
-
 impl Client {
-    pub async fn new(ctlr_tx: tokio::sync::mpsc::Sender<Event>) -> Result<Self, ipc::Error> {
+    pub async fn new() -> Result<(Self, ipc::ClientRead), ipc::Error> {
         tracing::debug!(
             client_pid = std::process::id(),
             "Connecting to IPC service..."
         );
-        let (mut rx, tx) = ipc::connect_to_service(ipc::ServiceId::Prod).await?;
-        let task = tokio::task::spawn(async move {
```

Comment:
So we're removing the (extra) thread from which we issue client IPC commands?

Reply:
We remove a task that was forwarding from the IPC stream to a channel. The other channel end and the IPC stream are both a `Stream` so after the refactoring of using `IpcServerMsg` directly, it is the same interface for downstream code: `Stream<Item = IpcServerMsg>`.